mdmon.c

```diff
@@ -394,9 +394,7 @@ int main(int argc, char *argv[])
 		/* launch an mdmon instance for each container found */
 		mdstat = mdstat_read(0, 0);
 		for (e = mdstat; e; e = e->next) {
-			if (e->metadata_version &&
-			    strncmp(e->metadata_version, "external:", 9) == 0 &&
-			    !is_subarray(&e->metadata_version[9])) {
+			if (is_mdstat_ent_external(e) && !is_mdstat_ent_subarray(e)) {
```

Comment:
`!is_mdstat_ent_subarray(e)` introduce redundancy by calling is_mdstat_ent_external(e) second time
so maybe just use `'is_subarray(ent->metadata_version + 9)`
unless you want to avoid using the magic number again ;)